### ISSUE TITLE
KeyError: 'name' when reading KGML from file

### ISSUE THREAD

**Issue by iquasere** [biopython/biopython#3466]
### Setup

I am reporting a problem with Biopython version, Python version, and operating
system as follows:

```
1.77

3.7.6 (default, Jan  8 2020, 19:59:22)
[GCC 7.3.0]

CPython
Linux-4.15.0-115-generic-x86_64-with-debian-buster-sid
```

### Expected behaviour

Read the file as a ```Bio.KEGG.KGML.KGML_pathway.Pathway``` object

### Actual behaviour

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/jsequeira/anaconda3/lib/python3.7/site-packages/Bio/KEGG/KGML/KGML_parser.py", line 38, in read
    pathway = next(pathways)
  File "/home/jsequeira/anaconda3/lib/python3.7/site-packages/Bio/KEGG/KGML/KGML_parser.py", line 71, in parse
    yield KGMLParser(elem).parse()
  File "/home/jsequeira/anaconda3/lib/python3.7/site-packages/Bio/KEGG/KGML/KGML_parser.py", line 172, in parse
    _parse_relation(element)
  File "/home/jsequeira/anaconda3/lib/python3.7/site-packages/Bio/KEGG/KGML/KGML_parser.py", line 154, in _parse_relation
    name, value = subtype.attrib["name"], subtype.attrib["value"]
KeyError: 'name'
```

### Steps to reproduce

```
kgml = KGML_parser.read(kegg_get("ko00680", "kgml"))
with open('map00680.xml','w') as f: f.write(kgml.get_KGML())
KGML_parser.read(open('map00680.xml','r'))
```

**Comment by noamkremen** [comment]
It might take me a while to test and submit PR, but this seems to fix it:
```
diff --git a/Bio/KEGG/KGML/KGML_pathway.py b/Bio/KEGG/KGML/KGML_pathway.py
index 0476f0bde..a75a185af 100644
--- a/Bio/KEGG/KGML/KGML_pathway.py
+++ b/Bio/KEGG/KGML/KGML_pathway.py
@@ -854,6 +854,9 @@ class Relation:
         }
         for (name, value) in self.subtypes:
             subtype = ET.Element("subtype")
-            subtype.attrib[name] = str(value)
+            subtype.attrib = {
+                "name": name,
+                "value": str(value)
+            }
             relation.append(subtype)
         return relation
```

**Comment by peterjc** [comment]
Cross reference #3465 - the timing suggests both issues stem from a KEGG update.

**Comment by peterjc** [comment]
@widdowquinn are you able to look at this and the proposed fix?

**Comment by widdowquinn** [comment]
Looking at it just now…

**Comment by widdowquinn** [comment]
I don't know when the change happened, but the new raw XML for a subtype is

```xml
    <relation entry1="35" entry2="71" type="ECrel">
        <subtype name="compound" value="33"/>
    </relation>
```

but `KGMLParser` writes

```xml
  <relation entry1="35" entry2="71" type="ECrel">
    <subtype compound="33"/>
  </relation>
```

and the proposed fix looks like it should work, to me.

**Comment by widdowquinn** [comment]
I think this might be fixed to be up to date with 0.7.2, now.

```python
>>> from Bio.KEGG.REST import kegg_get
>>> from Bio.KEGG.KGML import KGML_parser
>>> kgml = KGML_parser.read(kegg_get("ko00680", "kgml"))
>>> with open("map00680.xml", "w") as ofh:
...     ofh.write(kgml.get_KGML())
...     
... 
104280
>>> with open("map00680.xml", "r") as ifh:
...     roundtrip = KGML_parser.read(ifh)
...     
... 
>>> kgml.name
'path:ko00680'
>>> roundtrip.name
'path:ko00680'
>>> len(kgml.relations)
267
>>> len(roundtrip.relations)
267
```

**Comment by widdowquinn** [comment]
That should be a PR with a new test added (#3473)
